### ISSUE TITLE
Funding source [RFS]

### ISSUE THREAD

**Issue by jpullmann** [w3c/dxwg#66]
<section class='requirement' id='RFS'><h3>Funding source [<a href='https://www.w3.org/TR/dcat-ucr#RFS' target='_blank'>RFS</a>]</h3><p>Provide means to describe the funding (amount and source) of a Dataset (or entire Catalog).</p><br/><em>Related use cases</em>: <a href='https://www.w3.org/TR/dcat-ucr#ID31' target='_blank'>Modeling funding sources [ID31]</a> </section>

**Comment by nicholascar** [comment]
Should just cater for the source as a qualified Agent (prov:role :Funder) but will need an additional property/object for funding amount. Suggest using one of the (scientific) research projects ontology's patterns.

Also, ISO19115' RoleCodes (see [https://geo-ide.noaa.gov/wiki/index.php?title=ISO_19115_and_19115-2_CodeList_Dictionaries#CI_RoleCode](https://geo-ide.noaa.gov/wiki/index.php?title=ISO_19115_and_19115-2_CodeList_Dictionaries#CI_RoleCode)) include: funder & sponsor so we should align any role vocabs with those terms

**Comment by nicholascar** [comment]
Re-represented essentially as written as RDA Prov Patterns WG Use Case 45: [http://patterns.promsns.org/usecase/45](http://patterns.promsns.org/usecase/45)

**Comment by dr-shorthair** [comment]
See #71 
Funding sources are built into proposed Project ontology https://dr-shorthair.github.io/ont/project/ 
Is a direct link to the funding source needed in addition, or is the transitive link via link to context (Project) enough?

**Comment by nicholascar** [comment]
I think people will want to see direct links from a Dataset outwards within the context of DCAT, given its focus, however if a qualified Dataset/Agent relation with role Funder is made, then it should be mapped to a transitive link like you say. 

My main reason for saying this is that it's likely dataset catalogues that are not particularly graph-enabled will easily be able to more easily indicate Dataset -> Agent with a role Funder than a transitive link via a Project as they may not have Project objects represented as first class objects.

**Comment by smrgeoinfo** [comment]
long discussion on this in schema.org context https://github.com/schemaorg/schemaorg/issues/383
Also @danbri proposal for project in schema.org https://pending.webschemas.org/Project

**Comment by smrgeoinfo** [comment]
An important fact that needs to be associated with the dataset is an identifier for the funding instrument-- i.e. the grant number. Can a grant be a prov:Agent?
DataCite v4 model is a good template: a fundingReference (class) has properties funderName, funderIdentifier, awardNumber, and awardTitle.  Of course if there were a pattern for an award class, and the awardNumber was awardURI that would dereference to get a representation of the class, that's all you'd need.

**Comment by dr-shorthair** [comment]
In the research community I think 'Grant' is usually used to denote a kind of _Activity_, approximately equivalent to 'Project supported by a philanthropy or government funding agency or foundation'. 

In PROV-O terms the funder is an Agent associated with the Activity. In the [Project ontology](https://dr-shorthair.github.io/ont/project/#project-stakeholders) I have proposed `FundingAssociation` as a specialization of `prov:Association`, with the `Role` of the `agent` fixed to "funder", and an associated `SumOfMoney` and `grantNumber`.

**Comment by smrgeoinfo** [comment]
+1 Since (in NSF land...) a grant might have various co-PIs at different institutions, with different workplans (subprojects), it makes sense that the 'SumOfMoney' is on the association from a project to the grant. In my experience the major use case here is for funding agencies to be able to find all the products (datasets, services, software, publications) resulting from a particular grant.

**Comment by dr-shorthair** [comment]
In the [Project ontology](https://dr-shorthair.github.io/ont/project/#project-stakeholders) there is no class `Grant`. So it depends on whether by 'grant' you mean 

(a) the _project_ , or 

(b) a _funding-arrangement_ associated with a project. 

I think it is probably the latter, though the word 'grant' is often used synonymously with 'project' in the research community. 

Note also that `prov:Activity` is an occurrent, not a continuant. 
And `proj:FundingAssociation` is neither - it is a transitive sub-class of `prov:Influence`.

**Comment by makxdekkers** [comment]
My question would be whether the grant was specifically to produce the dataset, or was it funding the project/activity that produced the dataset? If it is the latter, the link to the grant would be at the level of the project, not of the dataset. 
About the funding amount, I would think that this is could be fairly complicated -- are grant amounts earmarked for individual datasets produced, what is the relation between real production/maintenance cost and funding amount, how are co-funding arrangements expressed?
Do we consider that this information is generally useful across domains, for example to allow searching for datasets that were funded by amounts in a certain range, or to distinguish between datasets that were funded by a grant and ones that weren't?
In my mind, we should leave this kind of information to the description of projects and activities, and not cover it in the description of the resulting datasets.

**Comment by andrea-perego** [comment]
@smrgeoinfo said:

> An important fact that needs to be associated with the dataset is an identifier for the funding instrument-- i.e. the grant number. Can a grant be a prov:Agent?
> DataCite v4 model is a good template: a fundingReference (class) has properties funderName, funderIdentifier, awardNumber, and awardTitle. Of course if there were a pattern for an award class, and the awardNumber was awardURI that would dereference to get a representation of the class, that's all you'd need.

I agree. It may be worth building upon what done in other communities and metadata schemas (as DataCite), where a similar discussion has already taken place, and for which they provided a solution.

A possible DCAT-based representation of the DataCite approach is available in the documentation to the DataCite / DCAT-AP mapping we did at JRC:

https://ec-jrc.github.io/datacite-to-dcat-ap/#mapping-funding-reference

Here is an example, based on the mappings defined above:

````turtle
a:Dataset a dcat:Dataset ;
  frapo:isFundedBy a:Project .

a:Project a foaf:Project ;
  dct:title "My Project" ;
  dct:identifier "123456" ;
  frapo:isAwardedBy a:Funder .
  
a:Funder a foaf:Organization ;
  foaf:name "My Funder" ;
  dct:identifier "78910" .  
````

The same could be written by using @dr-shorthair 's Project Ontology as follows (although I'm not sure it's correct):

````turtle
a:Dataset a dcat:Dataset ;
  frapo:isFundedBy a:Project .

a:Project a proj:Project ;
  dct:title "My Project" ;
  dct:identifier "123456" ;
  prof:hadSponsor a:Funder .
  
a:Funder a prov:Agent , foaf:Organization ;
  foaf:name "My Funder" ;
  dct:identifier "78910" .  
````

Personally, I think that what is in scope of DCAT are just the first 2 lines (whereas the rest can be included as an example on how to use external vocabularies):

````turtle
a:Dataset a dcat:Dataset ;
  frapo:isFundedBy a:Project .
````

so, IMHO, the main issue is whether we would like to use `frapo:isFundedBy` (or a similar property from existing vocabularies), rather than define a new one in the DCAT namespace.

PS: I see the discussion very much overlapping with #71, so I wonder whether we should try to merge them.

**Comment by makxdekkers** [comment]
I wonder whether the main characteristic is the funding relationship, or just that the Dataset was produced in the context of a project, funded or not. If it's more general, a different property could be better, e.g. foo:project.

**Comment by andrea-perego** [comment]
@makxdekkers said:

> I wonder whether the main characteristic is the funding relationship, or just that the Dataset was produced in the context of a project, funded or not. If it's more general, a different property could be better, e.g. foo:project.

In this case, it's specifically the funding relationship. For the one denoting the "project" that produced the dataset, there's the option of using `prov:wasGeneratedBy`.

**Comment by makxdekkers** [comment]
OK. It still feels a bit odd to me to state that a Dataset was funded by a project. It's rather that a Dataset was produced by or in the context of a project, which was funded in some way.

**Comment by andrea-perego** [comment]
@makxdekkers said:

> OK. It still feels a bit odd to me to state that a Dataset was funded by a project. It's rather that a Dataset was produced by or in the context of a project, which was funded in some way.

Yes, I agree. It's a bit stretched.

This is another option (using `schema:funder` instead of `frapo:isFundedBy`):

````turtle
a:Dataset a dcat:Dataset ;
  schema:funder a:Funder ;
  prov:wasGeneratedBy a:Project .

a:Funder a foaf:Organization ;
  foaf:name "My Funder" ;
  dct:identifier "78910" .

a:Project a prov:Activity . foaf:Project ;
  dct:title "My Project" ;
  dct:identifier "123456" ;
  schema:funder a:Funder .
````

**Comment by makxdekkers** [comment]
I see you duplicate the `schema:funder` in two places. I just wonder whether there is a strong case for such duplication -- for example, would a user really want to search for all Datasets funded by `a:Funder`? If so, it's OK with me.

**Comment by smrgeoinfo** [comment]
The upshot of the  extensive, multi-threaded conversation about this on the schema.org wiki (https://github.com/schemaorg/schemaorg/issues/383), led @danbri to[ propose several classes and properties there ](https://github.com/schemaorg/schemaorg/issues/383#issuecomment-398355608)
[Grant](https://webschemas.org/Grant)
[MonetaryGrant](https://webschemas.org/MonetaryGrant)
[Project](https://webschemas.org/Project)
[FundingAgency](https://webschemas.org/FundingAgency)
[FundingScheme](https://webschemas.org/FundingScheme)
[ResearchProject](https://webschemas.org/ResearchProject)
These are all in the 'pending' bin at Schema.org

I know that NSF program managers, or someone reviewing performance of a data management plan, would like to be able to search for all resources (datasets, services, software, publications....) that are products of e.g. NSF Award ICER 1639764 (s:Grant).   Legislative watch dogs might be very interested in finding all the products of a particular NIH, EPOS, or CSIRO program (s:FundingScheme), philanthropists might want to find all the products of a particular foundation (s:FundingAgency) or non-profit program (s:FundingScheme) to see what's actually getting published.  The kind of stuff that gets put in the acknowledgements section of just about any academic product.

Providing a link (resolvable identifier) to the actual funding instrument (award, gift, grant, bequest, cooperative agreement, contract, organization project number) would do the trick as long as there's some way to track from that link back to the rest of the administrative and organizational infrastructure behind the funding instrument.  I think that's the key piece of information.

the https://dcodings.github.io/DINGO ontology was developed by a working group that included @danbri in June, 2018; it is anchored in various schema.org classes, I think could be harmonized with @dr-shorthair 's model, and includes the classes and associations to represent linkage to funding scheme, agency, project etc. 
In the @dr-shorthair model, the linkage from a dataset to the grant identifier would be like
dcat:Dataset--> prov:wasGeneratedBy --> proj:Project -->proj:wasFundedThrough --> proj:FundingAssociation --> proj:grantNumber (the funder prov:Agent is also linked thru the FundingAssociation)

I was hoping for something more direct like dataset --> fundedBy --> grantNumber (and in the background grantNumber --> grant --> fundingScheme --> fundingAgency to get the rest of the details).   This puts the focus on the acknowledgment for support, perhaps not as elegant a model of the world.

**Comment by andrea-perego** [comment]
@makxdekkers said:

> I see you duplicate the `schema:funder` in two places. I just wonder whether there is a strong case for such duplication -- for example, would a user really want to search for all Datasets funded by `a:Funder`? If so, it's OK with me.

This is exactly one of the possible use cases.

I think it is important we consider both simple and complex scenarios. With simple ones, I mean catalogues which (compared to those in the complex scenario) are more "flat", in terms of the supported metadata schema, or that don't include the representation of projects/funders as standalone entities. In such cases, if they want to specify who is the funder and/or which is the project that created the dataset, both these pieces of information can only be attached to the dataset itself.

Another use of this reference to the funder from the dataset (in addition to the one between the project and the funder) is to re-enforce explicitly the indirect relationship dataset -> project -> funder.

**Comment by andrea-perego** [comment]
@nicholascar said:

> My main reason for saying this is that it's likely dataset catalogues that are not particularly graph-enabled will easily be able to more easily indicate Dataset -> Agent with a role Funder than a transitive link via a Project as they may not have Project objects represented as first class objects.

+1

**Comment by dr-shorthair** [comment]
Folks - we already agreed to use the property `prov:wasaGeneratedBy` to link from a `dcat:Dataset` to the activity that generated it, such as a project. See https://w3c.github.io/dxwg/dcat/#Property:dataset_wasgeneratedby 

So I think the remaining question is whether 
1. that's enough, and funding arrangements belong with the description of the project, (which may be available nearby, but probably not), or
2. we need an additional property to link a dataset directly to the funding arrangement. 

(We can also reconsider the decision around `wasGeneratedBy` of course.)

**Comment by andrea-perego** [comment]
@dr-shorthair , I personally don't question the use of `prof:wasGeneratedBy` to say that a given dataset is the output of a given project / activity. 

As you say, we need to understand whether the link to the funding should or not be attached directly with the dataset (which I personally think should be useful). I would also say that the question, in such a case, is whether such link should point to the funding agency or to the funded grant (as per @smrgeoinfo comment). In either cases, this property should be in addition to `prov:wasGeneratedBy`, and not replacing it.

BTW, I forgot that FOAF includes a property that can be used for that purpose, namely, `foaf:fundedBy` (http://xmlns.com/foaf/spec/#term_fundedBy).

**Comment by dr-shorthair** [comment]
Good. 

The [Project ontology](https://dr-shorthair.github.io/ont/project/#project-stakeholders) provides some clarity about the distinction- and relationships- between projects, funders, and grants. It takes a project-centric viewpoint, in which a 'grant' appears to correspond with the `FundingAssociation` class. 

A direct link from a _dataset_ to a (description of a) _grant_ would match the property-chain 
```
( prov:wasGeneratedBy  proj:wasFundedThrough ) 
```
A direct link from a dataset to the _funder_ would match 
```
( prov:wasGeneratedBy  proj:wasFundedThrough  prov:agent ) 
```
A direct link from a dataset to the _grant number_ would match 
```
( prov:wasGeneratedBy  proj:wasFundedThrough  proj:grantNumber ) 
```

**Comment by andrea-perego** [comment]
Just to note that the inclusion of a property for funders / funding agencies is also in scope with #76.

**Comment by dr-shorthair** [comment]
See also the proposal for [Qualified Relations](https://github.com/w3c/dxwg/wiki/Qualified-relations#agent-roles) - I  just added 'funder' to the worked example there. Following this pattern the link to the funder is through a property chain
```
( prov:qualifiedAttribution prov:agent )
```
_but with the constraint that the value of `dcat:hadRole`  is fixed_ to `<http://id.loc.gov/vocabulary/relators/fnd>`

**Comment by dr-shorthair** [comment]
Maybe http://id.loc.gov/vocabulary/relators/fnd

**Comment by andrea-perego** [comment]
Thanks, @dr-shorthair .

BTW, if we have enough evidence, I think we could also define a "funder" role in the DCAT namespace (`dcat:funder`).

**Comment by dr-shorthair** [comment]
OK - how about 

```
dcat:wasFundedBy
  rdf:type owl:ObjectProperty ;
  rdfs:comment "Party that provided funding to support the creation of this resource" ;
  rdfs:domain dcat:Resource ;
  rdfs:label "was funded by" ;
  owl:propertyChainAxiom (
      prov:wasGeneratedBy
      proj:wasFundedThrough
      prov:agent
    ) ;
.
```

with the expectation that the range is usually a `foaf:Agent` or `prov:Agent`

**Comment by nicholascar** [comment]
Here you have a funded *by* with a range of Agent and yet your proj property for an activity funding, same range Agent, is was funded *through*. Any reason to not have ...By in both cases?

*through* to me sounds like it should be used like this: funded through an on-going, perdurant, thing. An Agent is endurant so funded *By* seems better.

Summarising: can you align proj suffixes with the proposal here?

**Comment by dr-shorthair** [comment]
- funded _through_ has range `FundingAssociation` (i.e. a grant or contract) - this is just one element in the full path
- funded _by_ has range `Agent` - is shortcut for the full path

**Comment by andrea-perego** [comment]
@dr-shorthair , @nicholascar , just noting that, maybe, `dcat:funder` (in analogy with [`schema:funder`](http://schema.org/funder)) might more clearly suggest that the object is an agent.

**Comment by dr-shorthair** [comment]
```
dcat:funder
  rdf:type owl:ObjectProperty ;
  rdfs:comment "Party that provided funding to support the creation of this resource" ;
  rdfs:domain dcat:Resource ;
  rdfs:label "was funded by" ;
  rdfs:range [
      rdf:type owl:Class ;
      owl:unionOf (
          foaf:Agent
          prov:Agent
        ) ;
    ] ;
  owl:propertyChainAxiom (
      prov:wasGeneratedBy
      proj:wasFundedThrough
      prov:agent
    ) ;
  skos:closeMatch schema:funder ;
.
```

**Comment by miselico** [comment]
To give some context on the proposed schema.org , some of these were elaborated in a wikidata workshop. The outcome of which is presented here: https://github.com/shexSpec/schemas/blob/master/Wikidata/research_funding/

One issue I see is that the complexity of this topic will end up in it being rarely (correctly) used.

**Comment by makxdekkers** [comment]
It may be late in the day for my comment, but I do think that the indication of a funding source for a dataset as discussed here is quite domain-specific as it applies mostly to datasets that were produced in the context of a research grant. This may be a very real need in this particular domain, but I am not sure it needs to be in the core of DCAT -- especially because there are other ways to express the relationship (using prov and proj). I do not find @nicholascar's earlier statement '_as they may not have Project objects represented as first class objects_' a compelling argument to include a property in DCAT. 
In the work we did on the European DCAT-AP, in the context of government data portals, many people expressed the need for a property to link a dataset to a mandate/law/regulation that was the reason a  dataset had been created/published. However, it was thought that this was a domain-specific need and therefore not to be included in the core of DCAT.
Where do we draw the line between general and domain-specific requirements?

**Comment by dr-shorthair** [comment]
I have similar concerns to @makxdekkers about being tied too closely to one (admittedly important) application. As I showed above, the funder information is already available through two property chains:  

1. via a `proj:Project` instance 
```
( prov:wasGeneratedBy  proj:wasFundedThrough  prov:agent ) 
```

2. via a qualified relationship with an agent
```
( prov:qualifiedAttribution prov:agent )
```
when the value of dcat:hadRole is http://id.loc.gov/vocabulary/relators/fnd or http://registry.it.csiro.au/def/isotc211/CI_RoleCode/funder. 

But it is always a balance - common information should be easy to find, but how many roles should have their own predicates. We know that the old Dublin Core terms `creater`, `contributor` and `publisher` are not enough, so a scalable attribution facility is important. The MARC Relators and ISO CI_RoleCode vocabularies cover most bases, so perhaps that's enough? I guess it would be optimal if we were able to recommend one of these. While it is from the GI domain, the ISO list does have the advantage that it is not mixed in with a whole load of other stuff ...
